src/java/org/apache/cassandra/db/marshal/ByteArrayAccessor.java

```diff
@@ -236,6 +236,19 @@ public float toFloat(byte[] value)
         return ByteArrayUtil.getFloat(value, 0);
     }
 
+    @Override
+    public float[] toFloatArray(byte[] value, int dimension)
```

Comment:
for large vectors it is probably going to be faster to leverage FloatBuffer.get, but that can be a future optimization

Reply:
I agree. FWIW we are only using the `ByteBuffer` implementation atm and that does use `FloatBuffer.get`